nb-apis/nb-api/src/main/java/io/nosqlbench/nb/api/engine/metrics/reporters/PromPushReporterComponent.java

```diff
@@ -50,37 +50,54 @@ public class PromPushReporterComponent extends PeriodicTaskComponent {
     private final URI uri;
     private String bearerToken;
 
-    public PromPushReporterComponent(NBComponent parent, String endpoint, long intervalMs, NBLabels nbLabels) {
+    public PromPushReporterComponent(NBComponent parent, String endpoint,long intervalMs, NBLabels nbLabels, String prompushApikeyfile) {
         super(parent, nbLabels.and("_type", "prom-push"), intervalMs,  "REPORT-PROMPUSH",FirstReport.OnInterval, LastReport.OnInterrupt);
         String jobname = getLabels().valueOfOptional("jobname").orElse("default");
         String instance = getLabels().valueOfOptional("instance").orElse("default");
         if (jobname.equals("default") || instance.equals("default")) {
             logger.warn("It is highly recommended that you set a value for labels jobname and instance other than 'default'.");
         }
 
-        if (endpoint.matches("victoria:[a-zA-Z0-9._-]+:[0-9]+")) {
+        if (endpoint.matches("victoria:plain:[a-zA-Z0-9._-]+(:[0-9]+)?")) {
+            String[] parts = endpoint.split(":", 3);
+            endpoint = "http://" + parts[2] + "/api/v1/import/prometheus/metrics/job/JOBNAME/instance/INSTANCE";
+        } else if (endpoint.matches("victoria:tls:[a-zA-Z0-9._-]+(:[0-9]+)?")) {
+            String[] parts = endpoint.split(":", 3);
+            endpoint = "https://" + parts[2] + "/api/v1/import/prometheus/metrics/job/JOBNAME/instance/INSTANCE";
+        } else if (endpoint.matches("victoria:[a-zA-Z0-9._-]+(:[0-9]+)?")) {
+            System.out.println("match3");
```

Comment:
Either remove this debugging code or use a logger.

Reply:
Sorry, missed that. Removed!!